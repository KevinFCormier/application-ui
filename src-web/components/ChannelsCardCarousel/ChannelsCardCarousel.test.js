/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
// import { Provider } from 'react-redux'
// import store from '../../../src-web/index'
/* eslint-disable no-unused-vars */

const React = require('react')
const renderer = require('react-test-renderer')
const ChannelsCardCarousel = require('../ChannelsCardCarousel').default

describe('ChannelsCardCarousel', () => {
  const channelsCardData = [
    {
      name: 'Development',
      counts: {
        pending: {
          total: 3
        },
        'in progress': {
          total: 2
        },
        failed: {
          total: 1
        }
      }
    },
    {
      name: 'QA',
      counts: {
        pending: {
          total: 3
        },
        'in progress': {
          total: 2
        },
        failed: {
          total: 1
        }
      }
    },
    {
      name: 'Dev',
      counts: {
        pending: {
          total: 3
        },
        'in progress': {
          total: 2
        },
        failed: {
          total: 1
        }
      }
    }
  ]

  it('ChannelsCardCarousel renders correctly.', () => {
    // const tree = renderer
    //   .create(<Provider store={store}><ChannelsCardCarousel data={channelsCardData} /></Provider>)
    //   .toJSON()
    // expect(tree).toMatchSnapshot()
  })
})
