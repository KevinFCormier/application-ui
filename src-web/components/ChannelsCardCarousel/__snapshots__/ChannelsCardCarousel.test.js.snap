// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelsCardCarousel ChannelsCardCarousel renders correctly. 1`] = `
<div
  id="ChannelsCardCarousel"
>
  <div
    className="card-container-container"
  >
    <div
      className="individualCard"
    >
      <div
        className="card-container"
      >
        <div
          className="card-content"
        >
          <div
            className="card-name"
          >
            Channel
            : 
            Development
          </div>
          <div
            className="card-count-content"
          >
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  3
                </div>
              </div>
              <div
                className="card-status"
              >
                PENDING
              </div>
            </div>
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  2
                </div>
              </div>
              <div
                className="card-status"
              >
                IN PROGRESS
              </div>
            </div>
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  1
                </div>
              </div>
              <div
                className="card-status"
              >
                FAILED
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="individualCard"
    >
      <div
        className="card-container"
      >
        <div
          className="card-content"
        >
          <div
            className="card-name"
          >
            Channel
            : 
            QA
          </div>
          <div
            className="card-count-content"
          >
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  3
                </div>
              </div>
              <div
                className="card-status"
              >
                PENDING
              </div>
            </div>
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  2
                </div>
              </div>
              <div
                className="card-status"
              >
                IN PROGRESS
              </div>
            </div>
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  1
                </div>
              </div>
              <div
                className="card-status"
              >
                FAILED
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="individualCard"
    >
      <div
        className="card-container"
      >
        <div
          className="card-content"
        >
          <div
            className="card-name"
          >
            Channel
            : 
            Dev
          </div>
          <div
            className="card-count-content"
          >
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  3
                </div>
              </div>
              <div
                className="card-status"
              >
                PENDING
              </div>
            </div>
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  2
                </div>
              </div>
              <div
                className="card-status"
              >
                IN PROGRESS
              </div>
            </div>
            <div
              className="card-count-container"
              role="button"
              tabIndex="0"
            >
              <div
                className="card-count"
              >
                <div
                  className="card-count-total"
                >
                  1
                </div>
              </div>
              <div
                className="card-status"
              >
                FAILED
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
